### ISSUE TITLE
Support Euroscarf as source for plasmids

### ISSUE THREAD

**Issue by dgruano** [manulera/ShareYourCloning_backend#175]
I have worked with several plasmids coming from [Euroscarf](http://www.euroscarf.de). I think it would be good to support importing these plasmids as a source!

However, note that I opened one of the `.dna` files from Euroscarf in Share Your Cloning and it gave the error `Could not read the file XXX`, and the console displayed `Failed to load resource: the server responded with a status of 422 ()` ¨Method not allowed`.

I solved this by changing the extension from `.dna` to `.gb`. I am unsure whether we may encounter this problem with `.dna` files coming from other sources or whether SYC should check this in the background. Happy to open another issue if needed.

**Comment by manulera** [comment]
Thanks @dgruano, that's a great idea, and we can then contact them and maybe they can do an announcement if they have a mailing list or something similar.

You could do this integration similarly to the one from addgene (scraping). I don't think they have an API from what I can see.

A separate problem is calls to `/read_from_file` where the extension does not match the format. I opened a separate issue here: https://github.com/manulera/ShareYourCloning_frontend/issues/226

**Comment by dgruano** [comment]
I'm happy to make the integration. I also think that scraping is the way to go!

**Comment by manulera** [comment]
## Description

Euroscarf support

## Checklist

You can do some of these tasks in parallel, specially if you need to wait for approval, but it's good to keep in mind the intended order:

* [ ] Make a branch in [ShareYourCloning_LinkML](https://github.com/genestorian/ShareYourCloning_LinkML) where you implement the new source (see extra docs in there) on how to do this.
* [ ] Add your branch as a dependency (via commit id) using poetry in this repository, to update `shareyourcloning-linkml` dependency in `pyproject.toml`:
    ```bash
    poetry add git+hhttps://github.com/genestorian/ShareYourCloning_LinkML#<commit-id>
    ```
* [ ] Implement the new source in a branch in this repository. You will need to add a new endpoint in `main.py` and a new class in `pydantic_models.py` that will handle the new source. You can use the existing sources as a template. Note that you can add extra validation or methods (see examples as well).
* [ ] Write tests for the new source in `tests/test_endpoints.py`. You can use the existing tests as a template.
* [ ] Once the tests pass, merge the PR of the new source in [ShareYourCloning_LinkML] and make a release of the `shareyourcloning-linkml` package.
* [ ] Update the dependency in the branch of this repository repository to the new version.
    ```bash
    # Get version from https://pypi.org/project/shareyourcloning-linkml/
    # For instance, if the version is 0.1.6a0, then:
    poetry add shareyourcloning-linkml@0.1.6a0
    ```
* [ ] Implement the frontend functionality in a branch, following [the frontend wiki](https://github.com/manulera/ShareYourCloning_frontend/wiki/Checklist-%E2%80%90--adding-a-source).
* [ ] Make tests for the frontend functionality.
* [ ] Merge the PR in this repository into master.
* [ ]  Update the backend submodule in the frontend repository to the latest master.
* [ ]  Merge the PR in the frontend repository into master.
* [ ] You are done!
